packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -241,6 +243,12 @@ type SelectCommonProps = {
    * Modify size of select.
    */
   size?: 'small' | 'large';
+  /**
+   * Style of the select.
+   * Use 'borderless' to hide outline.
+   * @default 'default'
+   */
+  styleType?: 'default' | 'borderless';
```

Comment:
Should we make it more restrictive?
- We could allow borderless only for native select
- We could require passing a default selected value, and disallow placeholder

We can always make it less restrictive later if we learn about more use cases.

Reply:
Some more thoughts on placeholder vs default selected value: Placeholder text can be confusing and is generally bad from an accessibility/usability perspective ([read more](https://www.smashingmagazine.com/2018/06/placeholder-attribute/)). In the specific use case that this PR is trying to address, there is no visible label either, which adds to the potential confusion. The default selected value is what provides the necessary context.

![](https://github.com/iTwin/iTwinUI/assets/9084735/707dfa82-2844-4605-aa75-e4fed08b6313)

Reply:
After our discussion about this, I am leaning towards requiring a value to be selected.  This removes the issue of https://github.com/iTwin/iTwinUI/pull/1886#discussion_r1507946815 .

Reply:
Passing a `defaultValue` is now required in `CustomSelect` when `styleType="borderless"`. This is enforced in types. Additionally, if the user does not pass a `defaultValue`, the first option is automatically chosen.

Reply:
When `styleType="borderless"` in:

* `CustomSelect`: `defaultValue` is required (first option automatically selected if not passed) and `placeholder` is not allowed (ignored if passed). 
* `NativeSelect`: `defaultValue` is not required since anyways the first option was automatically selected when no placeholder or defaultValue was passed. So, I just disallowed `placeholder` (ignored if passed).

Reply:
That will simplify things and might also not cause the type issues (in the GitHub gist) that I mentioned in the PR description.

I hadn't done this so far because Jon didn't say so in his reply. Also, the CSS styles also handled borderless for the `CustomSelect` case. So, I thought we all agreed to support it for `CustomSelect` too.

@FlyersPh9 Can/Should we support `borderless` only on `NativeSelect`?

Reply:
Btw you might want to move the PR to draft next time until it is ready. I finished my review before seeing your updated PR description 😅

<img width="356" alt="image" src="https://github.com/iTwin/iTwinUI/assets/9084735/b531e885-a15f-4077-a46b-79828d5de368">

Reply:
Ohh, sounds good, will keep that in mind. Since I didn't update the PR description, I thought it was assumed that I'm working on the React portion and it's not complete and so not ready for review. But I'll mark the PR back to a draft next time to avoid confusion.

To confirm, the PR is now ready to review 🙂.

Reply:
After a short internal discussion, we decided to only support borderless for the native select (`<select>`) and not the custom select (`<div role="combobox">`).